source/Cosmos.Core_Asm/Delegate/DelegateInvokeAsm.cs

```diff
@@ -33,7 +35,7 @@ public override void AssembleNew(Assembler aAssembler, object aMethodInfo)
             XS.ClearInterruptFlag();
 
             XS.Comment("Get Invoke list count");
-            var xGetInvocationListMethod = typeof(MulticastDelegate).GetMethod("GetInvocationList");
+            var xGetInvocationListMethod = TypeOf(typeof(MulticastDelegate)).Methods.Single(m => m.Name == "GetInvocationList");
```

Comment:
Please dont use LINQ here, but a GetMethod(string) thing, like the original. That one oculd them be implemented by dictionary lookup, instead of enumerating the list.

Reply:
`GetMethod` right now is a simple `foreach` which returns when it finds a match. It's not good for performance, but I think that we shouldn't think too much about this for now. Also, there shouldn't be many cases where we want to get a specific member by name, so I'm not sure if a `Dictionary<string, MethodInfo>` would make the whole compilation faster...
I also have some plans to improve performance, but for now I tried to keep it simple (although it's a bit slower, which is somehow understandable as the core libraries used to be loaded by the runtime and we were compiling them, and now the compilation is completely isolated).

Reply:
Sure, but having the GetMethod do the foreach means we have it at 1 spot which we can easily optimize later..